pkg/kubelet/kuberuntime/kuberuntime_container_windows.go

```diff
@@ -109,6 +109,17 @@ func (m *kubeGenericRuntimeManager) generateWindowsContainerConfig(container *v1
 		wc.Resources.MemoryLimitInBytes = memoryLimit
 	}
 
+	// Windows containers use ephemeral storage by default and are created with
+	// a default volume size of 20GB.
+	// If specified - resources.limits.ephemeral-storage value will be used to
+	// override this default.
+	// https://docs.microsoft.com/virtualization/windowscontainers/manage-containers/container-storage#scratch-space
+	// https://docs.microsoft.com/virtualization/windowscontainers/manage-containers/container-storage#storage-limits
+	ephemeralStorageLimit := container.Resources.Limits.StorageEphemeral().Value()
```

Comment:
Reading through https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/

```
If the kubelet is managing local ephemeral storage as a resource, then the kubelet measures storage use in:

emptyDir volumes, except tmpfs emptyDir volumes
directories holding node-level logs
writeable container layers

If a Pod is using more ephemeral storage than you allow it to, the kubelet sets an eviction signal that triggers Pod eviction.
```

It seems that kubelet tracks more than just the rootfs with the Ephemeral Storage Limit value.  The RootfssizeinBytes seems to be the "writable layer" here, but would this mean pods would be evicted if they added `emtpydir` volume as well as resize the rootsfs  of the container?

Reply:
@msau42 @ddebroy @mauriciopoppe - Do any of you (or anyone else from SIG-storage) have any thoughts here?

Reply:
I think I am struggling a little bit with overloading the `ephemeralstoragelimit` concept and the `RootfsSizeInBytes`. I am not sure they are 1 to 1 but this is the first time I am looking at it.

I was looking around and it doesn't look like Linux sets the scratch layer sizes anywhere.  Does linux have a similar concept or is this windows only?

Reply:
cc @jingxu97

Reply:
The vhd itself for the volume is thin provisioned and will expand on data written, it's nowhere near 20GiB on the hosts disk

Reply:
> The vhd itself for the volume is thin provisioned and will expand on data written, it's nowhere near 20GiB on the hosts disk

That was what I thought - thanks for confirming!